https://salsa.debian.org/debian/keychain/-/blob/debian/master/debian/patches/empty-ssh-askpass.patch

Description: Handle an empty (or unset) SSH_ASKPASS variable
Bug-Debian: https://bugs.debian.org/325644
Forwarded: no
Author: Peter Pentchev <roam@debian.org>
Author: Tim Connors <twc@aaocbn.aao.gov.au>
Last-Updated: 2023-01-06

--- a/keychain.sh
+++ b/keychain.sh
@@ -1420,6 +1420,9 @@
 			IFS="$old_IFS"	# restore IFS
 			set +f			# re-enable globbing
 
+			if [ -z "$SSH_ASKPASS" ]; then
+				SSH_ASKPASS="$(command -v ssh-askpass || true)"
+			fi
 			if $noguiopt || [ -z "$SSH_ASKPASS" -o -z "$DISPLAY" ]; then
 				unset DISPLAY		# DISPLAY="" can cause problems
 				unset SSH_ASKPASS	# make sure ssh-add doesn't try SSH_ASKPASS
